src/Message/PurchaseRequest.php

```diff
@@ -69,60 +115,93 @@ public function getData()
+            'amount' => $this->getAmountInteger(),
+            'issuerid' => $this->getIssuer(),
+            'entrancecode' => $this->getEntranceCode(),
+            'description' => $this->getDescription(),
+            'including' => $this->getIncluding(),
+            'days' => $this->getDays(),
+            'returnurl' => $this->getReturnUrl(),
+            'cancelurl' => $this->getCancelUrl(),
+            'notifyurl' => $this->getNotifyUrl(),
+            'sha1' => $this->generateSignature(),
+            'testmode' => $this->getTestMode(),
         );
 
         /** @var \Omnipay\Common\CreditCard $card */
         $card = $this->getCard();
         if ($card) {
-            if ($this->getPaymentMethod() == 'overboeking' || $this->getPaymentMethod() == 'ecare') {
-                $data['billing_mail']       = $card->getEmail();
-                $data['billing_firstname']  = $card->getBillingFirstName();
-                $data['billing_lastname']   = $card->getBillingLastName();
+            if ($this->getPaymentMethod() == 'overboeking' || $this->getPaymentMethod() == 'klarna') {
+                $data['billing_mail'] = $card->getEmail();
+                $data['billing_firstname'] = $card->getBillingFirstName();
+                $data['billing_lastname'] = $card->getBillingLastName();
             }
 
-            if ($this->getPaymentMethod() == 'ecare') {
-                $data['billing_company']    = $card->getBillingCompany();
-                $data['billing_address1']   = $card->getBillingAddress1();
-                $data['billing_address2']   = $card->getBillingAddress2();
-                $data['billing_zip']        = $card->getBillingPostcode();
-                $data['billing_city']       = $card->getBillingCity();
-                $data['billing_country']    = $card->getBillingCountry();
-                $data['billing_phone']      = $card->getBillingPhone();
+            if ($this->getPaymentMethod() == 'klarna') {
+                $data['billing_company'] = $card->getBillingCompany();
+                $data['billing_address1'] = $card->getBillingAddress1();
+                $data['billing_address2'] = $card->getBillingAddress2();
+                $data['billing_zip'] = $card->getBillingPostcode();
+                $data['billing_city'] = $card->getBillingCity();
+                $data['billing_country'] = $card->getBillingCountry();
+                $data['billing_phone'] = $card->getBillingPhone();
+                $data['birthdate'] = date('dmY', strtotime($card->getBirthday()));
+                $data['makeinvoice'] = $this->getMakeInvoice();
+                $data['mailinvoice'] = $this->getMailInvoice();
+                $data['billing_countrycode'] = $this->getBillingCountrycode();
+                $data['shipping_countrycode'] = $this->getShippingCountrycode();
+
+                // only used for klarna account (required for klarna invoice as -1)
+                $data['pclass'] = - 1;
```

Comment:
Yes, unless we also want to add support for klarna-account, which I haven't had the time for to implement, perhaps someone else will want to pick that up based on this.